cea/analysis/sensitivity/sensitivity_demand_analyze.py

```diff
@@ -73,21 +59,57 @@ def analyze_sensitivity(samples_path):
         analysis_variables = ['mu_star', 'sigma', 'mu_star_conf']
         analysis_function = morris_analyze_function
 
+    # import The NumPy matrix containing the model inputs or samples
+    samples = np.load(os.path.join(samples_path, 'samples.npy'))
+    output_parameters = np.load(os.path.join(samples_path, 'output_parameters.npy'))
+    samples_count = len(samples)
+
+    # run the analysis for every input parameter
     for output_parameter in output_parameters:
-        # run the analysis for the given output parameter and write a worksheet for each analysis variable
-        analysis_results = []
-        for building in range(buildings_num):
-            # `Y` is a NumPy array containing the model outputs as used in the SALib analyze functions
-            Y = np.array([result.loc[building, output_parameter] for result in simulation_results])
-            result = analysis_function(problem, samples, Y)
-            analysis_results.append(result)
-
-        # write out a worksheet for each analysis result (e.g. 'S1', 'ST', 'ST_conf' for method == 'sobol')
-        for analysis_variable in analysis_variables:
-            worksheet_name = "%(output_parameter)s_%(analysis_variable)s" % locals()
-            building_results = [result[analysis_variable] for result in analysis_results]
-            pd.DataFrame(building_results, columns=problem['names']).to_excel(writer, worksheet_name)
-    writer.save()
+        # create excel writer
+        folder = os.path.join(samples_path, 'results')
+        if not os.path.exists(folder):
+            os.makedirs(folder)
+
+        if temporal_scale is 'yearly':
+            writer = pd.ExcelWriter(
```

Comment:
not sure how to implement this, could this be an issue of refactoring for the future?